airflow/hooks/dbapi.py

```diff
@@ -215,7 +226,7 @@ def run(self, sql, autocommit=False, parameters=None, handler=None):
         if handler is None:
             return None
 
-        if scalar:
+        if len(results) == 1:
```

Comment:
I feel instead of checking the length of `results`, we should instead check the length of `sql_statements`.

Also, is `scalar` no longer needed?